src/connection-manager/agreement.ts

```diff
@@ -195,6 +194,23 @@ class Agreement {
         last_ledger_time: new Date(),
       }
 
+      let server_version
+
+      if (isPreceedingFlagLedger(validation.ledger_index)) {
+        server_version = decodeServerVersion(validation.server_version)
+        const ballot: Ballot = {
+          signing_key,
+          ledger_index: Number(validation.ledger_index),
+          amendments: validation.amendments?.join(','),
+          base_fee: validation.base_fee ?? validation.ledger_fee?.fee_base,
```

Comment:
What is `validation.base_fee` as opposed to `validation.ledger_fee.fee_base`?

Reply:
A validation would only have a `base_fee` field if it wants to change the current fee. Otherwise it would not be included, so we use the fee from the network (i.e retrieved from the ledger) to fill in ones that does not vote for a change